Dockerfile.static-libs

```diff
@@ -0,0 +1,90 @@
+RUN /root/.cargo/bin/rustup target add $(uname -m)-unknown-linux-musl
+
+RUN wget https://github.com/libunwind/libunwind/releases/download/v1.3.1/libunwind-1.3.1.tar.gz && \
+    tar -zxf libunwind-1.3.1.tar.gz && \
+    cd libunwind-1.3.1/ && \
+    ./configure --disable-minidebuginfo --enable-ptrace --disable-tests --disable-documentation && make && make install
+
+COPY third_party/rustdeps /opt/rustdeps
+
+WORKDIR /opt/rustdeps
+
+RUN RUSTFLAGS="-C target-feature=+crt-static" /root/.cargo/bin/cargo build --release --target $(uname -m)-unknown-linux-musl
+RUN mv /opt/rustdeps/target/$(uname -m)-unknown-linux-musl/release/librustdeps.a /opt/rustdeps/librustdeps.a
+
+#        _
+#       | |
+#  _ __ | |__  _ __  ___ _ __  _   _
+# | '_ \| '_ \| '_ \/ __| '_ \| | | |
+# | |_) | | | | |_) \__ \ |_) | |_| |
+# | .__/|_| |_| .__/|___/ .__/ \__, |
+# | |         | |       | |     __/ |
+# |_|         |_|       |_|    |___/
+
+FROM php:7.3-fpm-alpine3.13 as phpspy-builder
+RUN apk add --update alpine-sdk
+RUN git clone https://github.com/pyroscope-io/phpspy.git && cd phpspy && git checkout 024461fbba5130a1dc7fd4f0b5a458424cf50b3a
+RUN cd phpspy && make CFLAGS="-DUSE_DIRECT"
+
+#              _
+#             | |
+#   __ _  ___ | | __ _ _ __   __ _
+#  / _` |/ _ \| |/ _` | '_ \ / _` |
+# | (_| | (_) | | (_| | | | | (_| |
+#  \__, |\___/|_|\__,_|_| |_|\__, |
+#   __/ |                     __/ |
+#  |___/                     |___/
+
+FROM golang:1.15.1-alpine3.12 as go-builder
+
+RUN apk add --no-cache make git zstd gcc g++ libc-dev musl-dev bash
+
+WORKDIR /opt/pyroscope
+
+RUN mkdir -p /opt/pyroscope/third_party/rustdeps/target/release
+COPY --from=rust-builder /opt/rustdeps/librustdeps.a /opt/pyroscope/third_party/rustdeps/target/release/librustdeps.a
+COPY third_party/rustdeps/rbspy.h /opt/pyroscope/third_party/rustdeps/rbspy.h
+COPY third_party/rustdeps/pyspy.h /opt/pyroscope/third_party/rustdeps/pyspy.h
+COPY third_party/phpspy/phpspy.h /opt/pyroscope/third_party/phpspy/phpspy.h
+COPY --from=phpspy-builder /var/www/html/phpspy/libphpspy.a /opt/pyroscope/third_party/phpspy/libphpspy.a
+COPY Makefile ./
+COPY tools ./tools
+COPY go.mod go.sum pyroscope.go ./
+RUN make install-dev-tools
+
+COPY pkg ./pkg
+COPY cmd ./cmd
+COPY scripts ./scripts
+
+RUN make build-rbspy-static-library
+RUN make build-pyspy-static-library
+RUN make build-phpspy-static-library
+
+
+#   __ _             _   _
+#  / _(_)           | | (_)
+# | |_ _ _ __   __ _| |  _ _ __ ___   __ _  __ _  ___
+# |  _| | '_ \ / _` | | | | '_ ` _ \ / _` |/ _` |/ _ \
+# | | | | | | | (_| | | | | | | | | | (_| | (_| |  __/
+# |_| |_|_| |_|\__,_|_| |_|_| |_| |_|\__,_|\__, |\___|
+#                                           __/ |
+#                                          |___/
+
+FROM scratch AS lib-exporter
+COPY --from=go-builder /opt/pyroscope/out/libpyroscope.rbspy.a /
+COPY --from=go-builder /opt/pyroscope/out/libpyroscope.pyspy.a /
```

Comment:
afaik, there is no way in Python to load a static lib, so I guess we have to go with dynamic lib.